### PR TITLE
[BEEEP] [PM-10132] Upgrade storybook to v8

### PR DESCRIPTION
**Pull request by Hinton**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10132

## 📔 Objective

Upgrade storybook to version v8 which is a major upgrade. Storybook provides an upgrade wizard which did most of the work.

- Ran npx storybook upgrade.
- Manually updated `remark-gfm` since the newer mdx requires v 4.
- Migrated all old stories still using `Story` to `StoryObj`.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .storybook/main.ts
Comment (Hinton) [inline .storybook/main.ts:1]: Storybook upgrade detected our monorepo and made these changes. I don't think they are strictly needed, but I don't think there's any harm keeping them either.

**Comment by Hinton** [comment]
@vleague2 @willmartian Can either of you review this first, and then mark it as ready for review to trigger the other teams?

File: apps/web/src/app/vault/components/vault-items/vault-items.stories.ts [hunk 1]
```diff
@@ -118,138 +118,136 @@ export default {
...
-  ciphers,
-  collections: [],
-  showOwner: true,
-  showCollections: false,
-  showGroups: false,
-  showPremiumFeatures: true,
-  showBulkMove: true,
-  showBulkTrashOptions: false,
-  useEvents: false,
-  cloneableOrganizationCiphers: false,
```
Comment (vleague2) [inline apps/web/src/app/vault/components/vault-items/vault-items.stories.ts:136]: ❓ Why take out this argument for these stories?

**Review by vleague2 — COMMENTED** [review]
Overall looks good to me. Just one question!

File: apps/web/src/app/vault/components/vault-items/vault-items.stories.ts (hunk 1 above)
Comment (willmartian) [reply apps/web/src/app/vault/components/vault-items/vault-items.stories.ts:136]: I am going to interpret this as a bad copy paste, will post a fix.

Comment (Hinton) [reply apps/web/src/app/vault/components/vault-items/vault-items.stories.ts:136]: `cloneableOrganizationCiphers` no longer exists on the component.

**Review by vleague2 — APPROVED** [review]

**Review by djsmith85 — APPROVED** [review]

**Review by r-tome — APPROVED** [review]

**Review by gbubemismith — APPROVED** [review]

**Review by Jingo88 — APPROVED** [review]

**Review by justindbaur — APPROVED** [review]

**Pull request merged**